spec/spec_helper.rb

```diff
@@ -59,9 +59,9 @@
   # The settings below are suggested to provide a good initial experience
   # with RSpec, but feel free to customize to your heart's content.
 
-  # Enable aggregate failures unless explcitly disabled at the spec level.
+  # Enable aggregate failures unless it's a feature spec or explcitly configured at the spec level.
   config.define_derived_metadata do |meta|
-    meta[:aggregate_failures] = true unless meta.key?(:aggregate_failures)
+    meta[:aggregate_failures] = (meta[:type] != :feature) unless meta.key?(:aggregate_failures)
```

Comment:
This doesn't seem to work for me.

I tested by placing this inside `spec/system/pages_spec.rb`:

```ruby
    expect(true).to eq(false)
    puts "!!!!!!!!!!!! got here"
```

And I see `!!!!!!!!!!!! got here` on the console when running specs.

Also I did some debugging and found that for a system test, `meta[:type] == :system`. But even if I change the line to use `:system` as follows:

```ruby
meta[:aggregate_failures] = (meta[:type] != :system) unless meta.key?(:aggregate_failures)
```

I still see `!!!!!!!!!!!! got here` on the console... 🤔

Reply:
Nice catch, I'll look in to it. I added this to CT and swore it worked, but I could be wrong. :)

Reply:
Confirmed: I was wrong. Will look in to why it doesn't work later.

Reply:
I used @mattbrictson's example to reproduce the problem, and I think what is happening is that in the check

> ```ruby
> meta[:aggregate_failures] = (meta[:type] != :system) unless meta.key?(:aggregate_failures)
> ```

the `aggregate_failures` key has somehow already been added to the metadata, regardless of whether it was set explicitly for an example or group. When I step through, I see this bit of code get hit several times when running one scenario, perhaps while rspec is internally building up metadata from the surrounding context.

I found a way to sidestep this by providing rspec filters to `define_derived_metadata`. TIL, thankfully [filters](https://relishapp.com/rspec/rspec-core/docs/hooks/filters) are pretty robust.